sdk/communication/communication-common/src/identifierModels.ts

```diff
@@ -181,3 +181,89 @@ export const getIdentifierKind = (
   }
   return { ...identifier, kind: "unknown" };
 };
+
+/**
+ * Returns the rawId for a given CommunicationIdentifier. You can use the rawId for encoding the identifier and then use it as a key in a database.
+ *
+ * @param identifier - The identifier to be translated to its rawId.
+ */
+export const getRawId = (identifier: CommunicationIdentifier): string => {
+  const identifierKind = getIdentifierKind(identifier);
+  switch (identifierKind.kind) {
+    case "communicationUser":
+      return identifierKind.communicationUserId;
+    case "microsoftTeamsUser": {
+      const { microsoftTeamsUserId, rawId, cloud, isAnonymous } = identifierKind;
+      if (rawId) return rawId;
+      if (isAnonymous) return `8:teamsvisitor:${microsoftTeamsUserId}`;
+      switch (cloud) {
+        case "dod":
+          return `8:dod:${microsoftTeamsUserId}`;
+        case "gcch":
+          return `8:gcch:${microsoftTeamsUserId}`;
+        case "public":
+          return `8:orgid:${microsoftTeamsUserId}`;
+      }
+      return `8:orgid:${microsoftTeamsUserId}`;
+    }
+    case "phoneNumber": {
+      const { phoneNumber, rawId } = identifierKind;
+      if (rawId) return rawId;
+      // strip the leading +. We just assume correct E.164 format here because validation should only happen server-side, not client-side.
+      return `4:${phoneNumber.replace(/^\+/, "")}`;
```

Comment:
why not substring instead of replace?

Reply:
Mainly because I wondered about a case where a phone number is accidentally supplied without the leading +. A substring of 1234 (instead of correct +1234) would then lead to a 4:234.
On the other hand maybe it's good if it breaks when the + is missing. In words I wanted to "remove the leading plus" and the replace felt like a truer representation of that.

Reply:
+1 to the already documented reasoning that client code should be tolerant in this case.